tool/planet/etcd.go

```diff
@@ -412,23 +523,100 @@ func getBaseEtcdDir(version string) string {
 }
 
 func etcdRestore(file string) error {
-	ctx := context.TODO()
-	log.Info("Restoring backup to temporary etcd")
+	log.Info("Initializing new etcd database")
+
+	datadir, err := getEtcdDataDir()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// To be re-entrant, shutdown the etcd server if its already running
+	err = etcdDisable(etcdUpgradeService, stopApiserverFalse)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// For the restore operation to be re-entrant, delete the data directory in-case the step is being re-run
+	err = os.RemoveAll(datadir)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = etcdEnable(etcdUpgradeService, "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	etcdConf := etcdconf.Config{
+		Endpoints: []string{DefaultEtcdUpgradeEndpoints},
+		KeyFile:   DefaultEtcdctlKeyFile,
+		CertFile:  DefaultEtcdctlCertFile,
+		CAFile:    DefaultEtcdctlCAFile,
+	}
+	client, err := etcdConf.NewClient()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// wait for the temporary etcd instance to complete startup
+	log.Info("Waiting for etcd initialization to complete")
+	err = waitEtcdHealthyTimeout(context.TODO(), 2*time.Minute, client)
```

Comment:
There's a share of arbitrary timeouts - do we want to keep that in a configuration or update on a need by need basis?

Reply:
So I'm generally of the position that because this is called by machine code, that this should simply be embedded at the call site. We don't really have a configuration system in place to allow overrides in a consistent fashion by setting some sort of background configuration file. If that were the case, then I would consider this for inclusion.